### ISSUE TITLE
Incompatible with elasticsearch-model

### ISSUE THREAD

**Issue by jcw-** [zdennis/activerecord-import#592]
The `import` aliases defined by this gem prevent the `import` method from `elasticsearch-model` from being added to a project's active record models.

Why? Because with `elasticsearch-model`, you cause an `import` method to be added to your model by using an `include`. However, note that `include` will not override or replace existing methods, if they already exist, they are simply skipped.

This means that since the `import` alias from `activerecord-import` is already in place when the model file does the `include`, the elastic search's `import` is never added. https://github.com/elastic/elasticsearch-rails/blob/master/elasticsearch-model/lib/elasticsearch/model/importing.rb#L119

This cannot be solved by gem load order - the include mechanism happens when the model files are loaded, which happens after the gems.

One workaround I found was to remove the import aliases in an initializer like this:
```
# Avoid a conflict with the `elasticsearch-model` gem by removing the `import`
# aliases created by the `activerecord-import` gem
ActiveRecord::Associations::CollectionAssociation.class_eval { remove_method :import }
ActiveRecord::Associations::CollectionProxy.class_eval { remove_method :import }
class ActiveRecord::Base
  class << self
    remove_method :import
    remove_method :import!
  end
end
```

Any chance the import aliases can be removed/dropped officially, or somehow made possible to easily "opt out" of this behavior?

**Comment by jkowens** [comment]
Thanks for posting this workaround for others to use if they run into this problem. I don't think we can drop the import alias as of right now, maybe it is something we could gradually work towards.

I'm definitely open to suggestions on how we might provide the option to opt out of the import alias. If anyone has thoughts please share!

**Comment by jkowens** [comment]
Can you still call the elastic search import using the `__elasticsearch__` proxy? For example
```
Article.__elasticsearch__.import
```
See:

https://github.com/elastic/elasticsearch-rails/issues/559

https://github.com/elastic/elasticsearch-rails/blob/master/elasticsearch-model/README.md#the-__elasticsearch__-proxy

**Comment by jcw-** [comment]
@jkowens Yes, it would appear so - making that another viable workaround, probably even the preferred one actually - thanks for pointing that out!

As far as options, a pattern I've seen before for gems is to provide something that can be set in an initializer:
https://github.com/carrierwaveuploader/carrierwave#configuring-carrierwave
https://thoughtbot.com/blog/mygem-configure-block
https://www.skcript.com/svr/the-easiest-configuration-block-for-your-ruby-gems/

**Comment by sushant12** [comment]
@jkowens did u get time to work on this?
